lib/redis/errors.rb

```diff
@@ -14,6 +14,10 @@ def initialize(reply_type)
     end
   end
 
+  # Raised when the response from the server is unparsable
+  class InvalidResponse < BaseError
+  end
+
```

Comment:
I wanted to use ProtocolError initially, but it's constructor is customized. I can refactor it though.